sdk/tools/kbdtool/parser.c

```diff
@@ -146,8 +146,8 @@ getVKNum(IN PCHAR p)
     if ((*p == '0') && ((*(p + 1) == 'x') || (*(p + 1) == 'X')))
     {
         /* Get the key number from the hex string */
-        *(p + 1) = 'x';
-        if (sscanf(p, "0x%x", &KeyNumber) == 1) return KeyNumber;
+        if (sscanf(p + 2, "%x", &KeyNumber) == 1 && KeyNumber <= 0xFF)
+            return KeyNumber;
```

Comment:
```suggestion
            return (UCHAR)KeyNumber;
```

Reply:
would it make sense here to do the cast proposed by Hermes, and stip the last condition of the if in return? Or would that be wrong?

Reply:
so you say your intent is to actively filter everything >= 0xFF and not return that as a KeyNumber. Why?